cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -547,7 +563,7 @@ Status TransferBinary(RecordReader* reader, MemoryPool* pool,
                       std::shared_ptr<ChunkedArray>* out) {
   if (reader->read_dictionary()) {
```

Comment:
So still when read_dictionary:
1. If writer set dictionary, it uses writers type
2. Otherwise, use int32 dict?

Reply:
That is indeed the intention here. In the case where the writer doesn't set the dictionary type, `TransferBinary` will call `TransferDictionary` [here](https://github.com/apache/arrow/blob/d88ef57a737b246c93e07ff98c19ff488bca1c59/cpp/src/parquet/arrow/reader_internal.cc#L549) which still hardcodes the `int32` type.